addons/mail/static/src/web/message_patch.js

```diff
@@ -65,9 +65,9 @@ patch(Message.prototype, "mail/web", {
                 return format.char(trackingValue.value);
             case "date":
                 if (trackingValue.value) {
-                    return luxon.DateTime.fromISO(trackingValue.value, { zone: "utc" })
-                        .setZone("system")
-                        .toLocaleString({ locale: this.userService.lang.replace("_", "-") });
+                    return luxon.DateTime.fromISO(trackingValue.value).toLocaleString({
+                        locale: this.userService.lang.replace("_", "-"),
+                    });
```

Comment:
I would follow the same LOCs as with `datetime`:

```is
const value = trackingValue.value
    ? deserializeDate(trackingValue.value)
    : trackingValue.value;
return formatters.get("date")(value);
```

That way it's the exact same computation, with the sole difference of `date` vs `datetime`.

I see that `deserializeDate` and `formatDate` (in 17.0) lack `setZone("default")` that `datetime` has. Probably another issue that can be fixed in this PR.

Reply:
Hello @alexkuhn , thanks for the check!
Indeed, `deserializeDate` do the right computation :+1: 

Regarding the lack of `setZone("default")`, I think it's not necessary to make the changes since it's a `date`, right?
Also, I see that we already set the zone as `default` when creating the date object in `deserializeDate` so not sure if it makes sense to call `setZone` again...:
https://github.com/odoo/odoo/blob/b04482ed87db05b3fc2523a08ce7ca5818af578b/addons/web/static/src/core/l10n/dates.js#L455
What do you think?

PS: Really not good on timezones, so inspired by a previous code :sweat_smile: :
https://github.com/odoo/odoo/blob/0008067ee56e8227e8b27dc34f51caf5e0a33a26/addons/web/static/src/core/l10n/dates.js#L153-L159